concordium-consensus/src/Concordium/External.hs

```diff
@@ -297,6 +303,48 @@ migrateGlobalState dbPath logM = do
             (False, True) -> logM GlobalState LLWarning "Cannot migrate legacy database as 'blockstate.dat' is absent."
             _ -> return ()
 
+-- |The opaque type that represents a foreign (i.e., living in Rust) object. The
+-- purpose of this context is to enable consensus to signal important events to
+-- the Rust code. In particular it is currently used to inform the GRPC2 server
+-- that a new block has arrived, or that a new block is finalized.
+data NotifyContext
+
+-- |Type of the callback used to send notifications.
+type NotifyCallback =
+  -- |Handle to the context.
+  Ptr NotifyContext ->
+  -- |The type of event. Only 0 and 1 are given meaning.
+  --
+  --   - 0 for block arrived
+  --   - 1 for block finalized
+  Word8
```

Comment:
Since this is an FFI bridge I am not sure this is useful. This callback is only used internally in this module and nobody outside needs to use the value.

We also don't generally have newtype wrappers for values going through FFI. They would help in some cases probably (in particular the startConsensus function), but in this case I don't think it will matter.